storage/google/cloud/storage/blob.py

```diff
@@ -301,6 +301,37 @@ def public_url(self):
             quoted_name=quote(self.name.encode("utf-8")),
         )
 
+    @classmethod
+    def from_string(cls, uri, client=None):
+        """Get a constructor for blob object by URI.
+
+         :type uri: str
+         :param uri: The blob uri pass to get blob object.
+
+         :type client: :class:`~google.cloud.storage.client.Client` or
+                      ``NoneType``
+         :param client: Optional. The client to use.
+
+         :rtype: :class:`google.cloud.storage.blob.Blob`
+         :returns: The blob object created.
+
+         Example:
+            Get a constructor for blob object by URI..
+
+            >>> from google.cloud import storage
+            >>> from google.cloud.storage.blob import Blob
+            >>> client = storage.Client()
+            >>> blob = Blob.from_string("gs://bucket/object")
+         """
+        from google.cloud.storage.bucket import Bucket
+
+        scheme, netloc, path, query, frag = urlsplit(uri)
```

Comment:
We need to update `download_blob_to_file` to use this `from_string` method.

https://github.com/googleapis/google-cloud-python/blob/6a06a5f3b99ccd2f7f5f003d2da08b501bdf55d2/storage/google/cloud/storage/client.py#L400